### PR TITLE
Option to build only application specific utilities (e.g. GFS)

### LINKED ISSUE #776: Build GFS application needed utilities.
Presently, when UFS utils are built, all utilities are built.
In most applications (certainly in ops), not all utilities are needed.

An option to build GFS specific utilities will be added.
The default behaviour of building all utilities will be retained.  No change is expected.

### PR DESCRIPTION
**Pull request by aerorahul**
## DESCRIPTION OF CHANGES: 
This PR:
- brings in an option to build GFS specific utilities
- retains the default of building all utilities
- Allows triggering of the option `-DGFS=ON` through the `build_all.sh` script.

## TESTS CONDUCTED: 
There is no change to the source code.
Only specific utilities are built for the GFS.  This option will be exercised in the GFS application.

## DEPENDENCIES:
None

## DOCUMENTATION:
No new capabilities have been added.

## ISSUE: 
Fixes #776

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,40 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
...
+option(FVCOMTOOLS "Enable building fvcom_tools.fd" ON)
+option(GBLEVENTS "Enable building gblevents.fd" ON)
+option(CPLD_GRIDGEN "Enable building cpld_gridgen.fd" ON)
+option(WEIGHT_GEN "Enable building weight_gen.fd" ON)
+
+# Option to build application specific utilities
+option(GFS "Enable building GFS-only utilities" OFF)
+
+# When building the GFS, the following need not be built
+if(GFS)
```
Comment (GeorgeGayno-NOAA) [inline CMakeLists.txt:38]: One odd thing I noticed. If you don't use double quotes, the message prints with no spaces between the words.

**Comment by GeorgeGayno-NOAA** [comment]
@LarissaReames-NOAA and @JeffBeck-NOAA - you work more closely with the SRW app, which only uses a subset of ufs_utils programs. Will this work well for the SRW app?

File: CMakeLists.txt (hunk 1 above)
Comment (aerorahul) [reply CMakeLists.txt:38]: You are correct @GeorgeGayno-NOAA 
I have replaced them with double quotes in dfb285b

**Comment by JeffBeck-NOAA** [comment]
> @LarissaReames-NOAA and @JeffBeck-NOAA - you work more closely with the SRW app, which only uses a subset of ufs_utils programs. Will this work well for the SRW app?

@GeorgeGayno-NOAA, this would definitely work for the SRW App.  We could introduce a -DSRW=ON flag to accomplish that.  @aerorahul, if I provided a list of the UFS_UTILS utilities that the SRW App uses, would it be easy to include that flag in this PR?  Thanks!

**Comment by GeorgeGayno-NOAA** [comment]
> > @LarissaReames-NOAA and @JeffBeck-NOAA - you work more closely with the SRW app, which only uses a subset of ufs_utils programs. Will this work well for the SRW app?
> 
> @GeorgeGayno-NOAA, this would definitely work for the SRW App. We could introduce a -DSRW=ON flag to accomplish that. @aerorahul, if I provided a list of the UFS_UTILS utilities that the SRW App uses, would it be easy to include that flag in this PR? Thanks!

@JeffBeck-NOAA Let's handle the SRW App in a separate PR.

**Comment by JeffBeck-NOAA** [comment]
> @JeffBeck-NOAA Let's handle the SRW App in a separate PR.

Sounds good!

**Comment by GeorgeGayno-NOAA** [comment]
@aerorahul The lsm library is required by global_cycle (and no other programs). So if I do "-DLSM=OFF -DGCYCLE=ON" the build breaks. I suggest the following:

```
--- a/sorc/CMakeLists.txt
+++ b/sorc/CMakeLists.txt
@@ -10,9 +10,7 @@ if(SNOW2MDL)
 endif()
 if (OpenMP_FOUND)
   if(GCYCLE)
-  add_subdirectory(global_cycle.fd)
-  endif()
-  if(LSM)
+    add_subdirectory(global_cycle.fd)
     add_subdirectory(lsm_routines.fd)
   endif()
 else()
```

Then, get rid of the LSM variable.

**Comment by aerorahul** [comment]
> @aerorahul The lsm library is required by global_cycle (and no other programs). So if I do "-DLSM=OFF -DGCYCLE=ON" the build breaks. I suggest the following:
> 
> ```
> --- a/sorc/CMakeLists.txt
> +++ b/sorc/CMakeLists.txt
> @@ -10,9 +10,7 @@ if(SNOW2MDL)
>  endif()
>  if (OpenMP_FOUND)
>    if(GCYCLE)
> -  add_subdirectory(global_cycle.fd)
> -  endif()
> -  if(LSM)
> +    add_subdirectory(global_cycle.fd)
>      add_subdirectory(lsm_routines.fd)
>    endif()
>  else()
> ```
> 
> Then, get rid of the LSM variable.

LSM is included when global-cycle is requested.
Thanks for letting me know this piece of information.

**Comment by GeorgeGayno-NOAA** [comment]
I tried the new option (`-DGFS=ON`) and it worked. I also tried turning off the compilation of individual components (Ex: `-DCHGRES=OFF`) and it worked. The default compilation worked on Hera, Jet and Dogwood. 

Will approve and merge. Thanks for adding this useful option.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**